engines/cpu.c

```diff
@@ -63,14 +179,65 @@ static enum fio_q_status fio_cpuio_queue(struct thread_data *td,
 		return FIO_Q_BUSY;
 	}
 
-	usec_spin(co->cpucycle);
+	switch (co->cpumode) {
+		case FIO_CPU_NOOP:
+			usec_spin(co->cpucycle);
+			break;
+		case FIO_CPU_QSORT:
+			do_qsort(td);
+			break;
+	}
+
 	return FIO_Q_COMPLETED;
 }
 
+static int noop_init(struct thread_data *td)
+{
+	struct cpu_options *co = td->eo;
+
+	log_info("%s (noop): ioengine=%s, cpuload=%u, cpucycle=%u\n",
+		td->o.name, td->io_ops->name, co->cpuload, co->cpucycle);
+	return 0;
+}
+
+static int qsort_cleanup(struct thread_data *td)
+{
+	struct cpu_options *co = td->eo;
+	if (co->qsort_data)
+		free(co->qsort_data);
+	return 0;
+}
+
+static int qsort_init(struct thread_data *td)
+{
+	struct cpu_options *co = td->eo;
+
+	// Setting up a default entropy
+	mwc_t mwc = {
+	        (521288629UL),
+	        (362436069UL)
+	};
+
+	if ((co->qsort_data = calloc(qsort_size, sizeof(*co->qsort_data))) == NULL) {
```

Comment:
Separate lines for assignment and check of return

Reply:
ok, done.